### DISCUSSION TITLE
Can scopeSpans fields be automatically configured?

### DISCUSSION THREAD

**Discussion by huangtao730** [open-telemetry/opentelemetry-java-instrumentation#12970] [Q&A]
“ScopeSpans” does not have some fields I need  such as local ip. Can I get these field information through parameter configuration?Or do I need to modify the source code to get it?The structure of “ScopeSpans” is not what I want. Can I modify this structure to suit my needs?If so, what do I need to do?Tks

**Comment by steverao** [comment]
Can you get them from span in ScopeSpans? If current design can't meet your demand.
The definition of `ScopeSpans` comes from https://github.com/open-telemetry/opentelemetry-proto/blob/2bd940b2b77c1ab57c27166af21384906da7bb2b/opentelemetry/proto/trace/v1/trace.proto#L68-L83
If you have any suggestion or modification, you need to file an issue in related repo.

**Reply by huangtao730** [reply]
We are currently using skywalking and now want to change to opentelementry。So there are many compatibility issues that need to be considered。
I also can't get the fields I want from the "spans" in "ScopeSpans", such as the remote IP and port of the httpclient component.
Secondly, our traceId is generated according to a certain strategy. Can the generation logic of traceId here be modified?For example, this is our current traceId:"5a0a0a651df8d17351813345470620"。Bits 1 to 4 represent the process ID, bits 5 to 13 represent the machine IP code, bits 14 represent the environment code, bits 15 to 28 represent the timestamp, and the last 4 bits are an auto-increment number used for modulus sampling calculations.
Therefore, we need to modify the traceId generation specification to be compatible with our current old version of APM。
Any good solution? Tks！

**Reply by steverao** [reply]
> I also can't get the fields I want from the "spans" in "ScopeSpans", such as the remote IP and port of the httpclient component.
 
Can you elaborate where you want to get attributes like `remote IP` and `port` in span? In your application or elsewhere?

> Can the generation logic of traceId here be modified?

You can do it by using a custom `IdGenerator`, refer to https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension#extensions-examples

**Reply by huangtao730** [reply]
In opentelemetry-javaagent-2.11.0-SNAPSHOT.jar。When sending data to the collector, I need to put the field of "remote IP" in the span, but I don't get it in the span.
Also, I cannot obtain some other metrics, such as maximum time consumption, minimum time consumption, percentile value, number of exceptions, number of timeouts, etc.
I have some custom metrics, How do I get it?
